pkg/kubelet/dockershim/docker_sandbox.go

```diff
@@ -273,6 +313,30 @@ func (ds *dockerService) makeSandboxDockerConfig(c *runtimeApi.PodSandboxConfig,
 	return createConfig, nil
 }
 
+func (ds *dockerService) GetPodSandboxNetNS(podSandboxID string) (string, error) {
+	r, err := ds.client.InspectContainer(podSandboxID)
+	if err != nil {
+		return "", err
+	}
+
+	return getNetworkNamespace(r), nil
+}
+
+func (ds *dockerService) GetPodAnnotations(namespace, name, podSandboxID string) (map[string]string, error) {
+	r, err := ds.client.InspectContainer(podSandboxID)
+	if err != nil {
+		return nil, err
+	}
+	_, annotations := extractLabels(r.Config.Labels)
+	return annotations, nil
+}
+
+func (ds *dockerService) GetPodHostportMapping() ([]*hostport.PodPortMapping, error) {
+	//TODO: Extract container port mapping from docker container labels and constrcut PodPortMappings
```

Comment:
@freehan could you file a github issue for the followup work after this PR? Thanks!